DigitalLearningSolutions.Web/Controllers/MyAccountController.cs

```diff
@@ -131,6 +133,24 @@ private IActionResult EditDetailsPostSave(EditDetailsFormData formData, DlsSubAp
             if (!ModelState.IsValid)
             {
                 var model = new EditDetailsViewModel(formData, dlsSubApplication);
+                var validationResults = model.Validate(new ValidationContext(model, null, null));
+                foreach (var result in validationResults)
+                {
+                    var key = string.Join("", result.MemberNames);
+                    if (ModelState.TryGetValue(key, out var value))
+                    {
+                        if (value.ValidationState == ModelValidationState.Invalid)
+                        {
+                            continue;
+                        }
+                    }
+
+                    ModelState.TryAddModelError(
+                        key,
+                        result.ErrorMessage
+                    );
+                }

```

Comment:
Why is this necessary? As far as I can tell we haven't needed anything like this before.

Reply:
Since EditDetailsFormData has other data annotations, they take priority over IValidatableObject so when user tries to save the page without selecting the professional registration number and when there are other entries are not entered like first name, password etc. -- only list of data annotations validations will be shown

IValidatableObject validations will only be shown when all other validations are passed. To resolve this, we are calling validate method implicitly so all the validations are shown. There is check to avoid duplicate validation being added to ModelState.

If we remove this code, professional registration validation will only be shown when all other existing validations are passed

Reply:
Aha, I see the problem. Sounds like this could come up elsewhere then, is there any way to make this reusable? Maybe a helper function like `public static void ManuallyValidate(this IValidatableObject model, ModelStateDictionary ModelState)` that runs this exact code?

Reply:
I think, yes, we can do. I will make changes. Thanks

Reply:
I have moved it to helper class